src/geo/leaflet/leaflet-cartodb-vector-layer-group-view.js

```diff
@@ -22,18 +22,8 @@ var LeafletCartoDBLayerGroupView = L.CartoDBd3Layer.extend({
 
   _onTileJSONChanged: function () {
     var tilejson = this.model.get('urls');
-    this.setUrl(tilejson.tiles[0]);
-
     this.options.styles = this.model.layers.pluck('cartocss');
-
-    if (this.renderers.length === 0) {
-      L.CartoDBd3Layer.prototype.onAdd.call(this, this.leafletMap);
-    } else {
-      this.setProvider({
-        styles: this.model.layers.pluck('cartocss'),
-        urlTemplate: tilejson.tiles[0]
-      });
-    }
+    this.setUrl(tilejson.tiles[0]);
```

Comment:
IMO `setURL` is something that is particular of the XYZProvider we're using (although we are not setting this up explicitly), so I'd expect this to be `this.provider.setURL(...)`.